### DISCUSSION TITLE
Is there an API to set credentials in the cloud app dashboard?

### DISCUSSION THREAD

**Discussion by mkniewallner** [renovatebot/renovate#31306] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

At work, we have an automation that regularly refreshes some secrets in a git repository that hosts our shared Renovate configuration, for them to be available across the organisation. Now that secrets need to be added to the cloud app dashboard, this automation won't work anymore.

Is there an API for the developer dashboard that allows creating/updating secrets in the cloud app? For now, to update the secrets regularly, we could probably use `https://developer.mend.io/api/orgs/github/<org>/secrets/<secret>` that is used by the dashboard, but to authenticate, we would need to sign to the dashboard manually with a GitHub account, then retrieve `mend_session` cookie, which is only valid for a couple of days. If this is the only endpoint available, is there maybe another authentication method that would be more automation-friendly?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
There isn't currently any API available for the Mend backend

**Reply by viceice** [reply]
but probably makes sense to add. so users can use a GitHub pat as auth to that API. 🤗

**Reply by D3X** [reply]
We're using AWS CodeArtifact to host our private packages. Its access tokens are short-lived (12h max), so we can no longer use Renovate, unless we set up a proxy server with long-lived credentials.
We used to update the renovate config with encrypted token using a periodic task running in CI, but that's not viable anymore...

So, overall, +1 to the API access

**Reply by lbartolotti-gocity** [reply]
@viceice I'm in same situation of D3X, and could really use this API call. Is there an open issue for it already?

**Reply by viceice** [reply]
I'm not an employee of mend, so I don't know. 🤷‍♂️

**Reply by rarkins** [reply]
There is no issue for this, because it's unrelated to this open source CLI.

This will be prioritized once a _customer_ needs it. If you're a customer, feel free to raise this to your support manager. If you're not, feel free to become one.

**Reply by jacek-jablonski** [reply]
Same for GCP's Artifact Registry. Timed service account keys forces us to do manual rotation.
